src/main/scala/com/gu/zuora/ZuoraService.scala

```diff
@@ -27,24 +30,22 @@ class ZuoraService(config: ZuoraConfig, client: FutureHttpClient, baseUrl: Optio
   def getAccount(accountNumber: String): Future[GetAccountResponse] =
     get[GetAccountResponse](s"accounts/$accountNumber", authHeaders)
 
-  def getAccountIds(identityId: String): Future[List[String]] = {
-    val queryData = QueryData(s"select AccountNumber from account where IdentityId__c = '${identityId.toLong}'")
-    postJson[AccountQueryResponse](s"action/query", queryData.asJson, authHeaders).map(_.records.map(_.AccountNumber))
+  def getAccountFields(identityId: IdentityId): Future[List[DomainAccount]] = {
+    // WARNING constructing queries from strings is inherently dangerous.  Be very careful.
+    val queryData = QueryData(s"select AccountNumber, CreatedSessionId__c from account where IdentityId__c = '${identityId.value}'")
```

Comment:
the identity id is verified when it is put into the type.  I could re-add the toLong here for safety though?  Really for proper abstraction we need to use the SafeQueryBuilder class which basically escapes everything automatically https://github.com/guardian/zuora-auto-cancel/blob/master/lib/zuora/src/main/scala/com/gu/util/zuora/SafeQueryBuilder.scala